src/permissions/plugins/github/index.ts

```diff
@@ -0,0 +1,130 @@
+          throw error;
+        }
+      }
+
+      if (!environmentExists) {
+        builder.addContext(
+          `:sparkles: Creating GitHub environment \`${GITHUB_DEFAULT_ENVIRONMENT}\` for repository \`${repo.name}\``,
+        );
+        console.info(
+          chalk.green('Creating GitHub environment'),
+          chalk.cyan(GITHUB_DEFAULT_ENVIRONMENT),
+          'for repository',
+          chalk.cyan(repo.name),
+        );
+
+        if (!IS_DRY_RUN) {
+          await octokit.repos.createOrUpdateEnvironment({
+            owner: org,
+            repo: repo.name,
+            environment_name: GITHUB_DEFAULT_ENVIRONMENT,
+            deployment_branch_policy: null,
+          });
+        }
+      }
+
+      const { data: policies } = await octokit.repos.listDeploymentBranchPolicies({
+        owner: org,
+        repo: repo.name,
+        environment_name: GITHUB_DEFAULT_ENVIRONMENT,
+      });
+
+      const hasDefaultBranchPolicy = policies.branch_policies?.some(
+        (policy: any) => policy.name === GITHUB_DEFAULT_BRANCH,
+      );
+
+      if (!hasDefaultBranchPolicy) {
+        builder.addContext(
+          `:shield: Adding deployment branch policy for \`${GITHUB_DEFAULT_BRANCH}\` to \`${GITHUB_DEFAULT_ENVIRONMENT}\` environment in \`${repo.name}\``,
+        );
+        console.info(
+          chalk.green('Adding deployment branch policy for'),
+          chalk.cyan(GITHUB_DEFAULT_BRANCH),
+          'to',
+          chalk.cyan(GITHUB_DEFAULT_ENVIRONMENT),
+          'environment in',
+          chalk.cyan(repo.name),
+        );
+
+        if (!IS_DRY_RUN) {
+          await octokit.repos.createDeploymentBranchPolicy({
+            owner: org,
+            repo: repo.name,
+            environment_name: GITHUB_DEFAULT_ENVIRONMENT,
+            name: GITHUB_DEFAULT_BRANCH,
+            type: 'branch',
+          });
+        }
+        builder.addContext(
+          `:white_check_mark: Successfully configured \`${GITHUB_DEFAULT_ENVIRONMENT}\` environment for \`${repo.name}\``,
+        );
+        console.info(
+          chalk.green('Successfully configured'),
+          chalk.cyan(GITHUB_DEFAULT_ENVIRONMENT),
+          'environment for',
+          chalk.cyan(repo.name),
+        );
+      } else {
+        // Only log this, we don't want this in the Slack channel on each Sheriff run
+        console.info(
+          chalk.green('GitHub environment'),
+          chalk.cyan(GITHUB_DEFAULT_ENVIRONMENT),
+          'already configured for',
+          chalk.cyan(repo.name),
+        );
+      }
+    } catch (error: any) {
+      builder.addContext(
+        `:x: Failed to configure GitHub environment \`${GITHUB_DEFAULT_ENVIRONMENT}\` for \`${repo.name}\`: ${error.message}`,
+      );
+      console.error(
+        `Failed to configure GitHub environment ${GITHUB_DEFAULT_ENVIRONMENT} for ${repo.name}:`,
+        error.message,
+      );
+    }
+  };
+}
+
+export const githubPlugin = new GitHubPlugin();
```

Comment:
Can we call this the `trustedPublisherPlugin` or something instead, 99% of github stuff should go in run.ts, this is just a trusted publisher specific extension thingy